CHANGELOG.md

```diff
@@ -1,35 +1,68 @@
-     - OS incompatibility fixes (path separator).
-   - https://github.com/textX/textX/pull/98
-     - Added STRICTFLOAT as buildin type to allow to distinguish ints from
-       floats in NUMBERs. Fixed docu link.
-   - https://github.com/textX/textX/pull/93
-     - Changed attribute name for the metamodel object (from 
-       "metamodel._parser" to "metamodel._parser_blueprint").
-   - https://github.com/textX/textX/pull/120
-     Unicode requirement for (meta)-model strings API parameters made strict.
-     This should prevent common errors with Python 2.x
-     See:
-     https://github.com/textX/textX/issues/105
-     https://github.com/textX/textX/pull/99
-     https://github.com/textX/textX/issues/117
-
-
-* 2018-10-06 Release 1.8.0
+All _notable_ changes to this project will be documented in this file.
+
+The format is based on _[Keep a Changelog][keepachangelog]_, and this project
+adheres to _[Semantic Versioning][semver]_.
+
+Everything that is documented in the [official docs][textXDocs] is considered
+the part of the public API.
+
+Backward incompatible changes are marked with **(BIC)**. These changes are the
+reason for the major version increase so when upgrading between major versions
+please take a look at related PRs and issues and see if the change affects you.
+
+## [Unreleased]
+
+### Added
+
+  - Adding examples and documentation related to scope providers (related to
+    model modification through scope providers) ([#168])
+  - metamodel export feature for [PlantUML] ([#165])
+  - `textx_isinstance` from `textx.scoping.tools` made available in `textx`
+    ([#164], [#157])
+  - CLI extensibility ([#162], [#161])
+  - An initial version of FAQ page ([#138]). Thanks Aluriak@GitHub
+  - A version of `calc.py` exercises usage of
+    `text.scoping.tools.textx_isinstance()` to inspect model objects types
+    during traversal. ([#136], [#123]). Thanks dkrikun@GitHub
+  - A version of `calc.py` in expression example that exercises dynamically adding
+    properties to object classes ([#126]). Thanks dkrikun@GitHub
+  - python like imports (named import rules, scope providers affected) ([#114])
+  - Added `STRICTFLOAT` as buildin type to allow to distinguish ints from floats
+    in `NUMBER`. Fixed docu link ([#98]). Possible **(BIC)**
+  - Added [flake8] and [coverage] checking ([#92])
+
+### Changed
+
+  - CLI migrated to the [click] library ([#162])
+  - Docs improvements ([#146], [#153], [#151]). Thanks simkimsia@GitHub.
+  - `FQN` constuctor param `follow_loaded_models` removed and introduced
+    callback `scope_rediction_logic` to implement arbitrary logic on FQN
+    resolution ([#133], [#114], [#103]) **(BIC)**
+  - Changed attribute name for the meta-model object (from `metamodel._parser`
+    to `metamodel._parser_blueprint`). ([#93]) **(BIC)**
+  - Started using _[Keep a Changelog][keepachangelog]_ ([#?])
+  - Started using _[Semantic Versioning][semver]_ ([#?])
```

Comment:
References with '?'

Reply:
Good catch! Yeah, I put a question mark with an idea to replace with a proper PR number as soon as I make one, but forgot to update. Will do.